### PR TITLE
task/34034: Login

### PR DESCRIPTION
**Pull request by TouanAin**

### CODE REVIEW

File: app/assets/javascripts/sessions.coffee [hunk 1]
```diff
@@ -0,0 +1,3 @@
+# Place all the behaviors and hooks related to the matching controller here.
+# All this logic will automatically be available in application.js.
+# You can use CoffeeScript in this file: http://coffeescript.org/
```
Comment (phamkykhoi) [inline app/assets/javascripts/sessions.coffee]: Remove this file if not use

File: app/assets/stylesheets/sessions.scss [hunk 2]
```diff
@@ -0,0 +1,3 @@
+// Place all the styles related to the Sessions controller here.
+// They will automatically be included in application.css.
+// You can use Sass (SCSS) here: http://sass-lang.com/
```
Comment (phamkykhoi) [inline app/assets/stylesheets/sessions.scss]: Remove this file if not use

File: app/controllers/sessions_controller.rb [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+  end
+
+  def create
+    user = User.find_by(email: params[:session][:email].downcase)
+    if user && user.authenticate(params[:session][:password])
+      log_in user
+      params[:session][:remember_me] == '1' ? remember(user) : forget(user)
+      redirect_to user
+    else
+      flash.now[:danger] = 'Invalid email/password combination'
```
Comment (phamkykhoi) [inline app/controllers/sessions_controller.rb]: I18n

File: app/models/user.rb [hunk 4]
```diff
@@ -8,4 +8,31 @@ class User < ActiveRecord::Base
...
+  
+  # Returns a random token.
+  def User.new_token
+    SecureRandom.urlsafe_base64
+  end
+
+  # Remembers a user in the database for use in persistent sessions.
+  def remember
+    self.remember_token = User.new_token
+    update_attribute(:remember_digest, User.digest(remember_token))
```
Comment (chuanhtuan) [inline app/models/user.rb]: use update_attribute instead of update_attribute

File: app/models/user.rb [hunk 5]
```diff
@@ -8,4 +8,31 @@ class User < ActiveRecord::Base
...
+  end
+
+  # Returns true if the given token matches the digest.
+  def authenticated?(remember_token)
+    BCrypt::Password.new(remember_digest).is_password?(remember_token)
+  end
+
+  # Forgets a user.
+  def forget
+    update_attribute(:remember_digest, nil)
```
Comment (chuanhtuan) [inline app/models/user.rb]: use update_attribute instead of update_attribute

**Pull request merged**